### PR TITLE
fix(CollectionLabel table): [BACK-1653] Refactor table to add collection and label associations

### PR DESCRIPTION
**Pull request by Herraj**
## Goal

Follow up PR to https://github.com/Pocket/collection-api/pull/996.

We missed the two table associations to `Collection` and `Label` table. Also adding a composite key and a `createdAt` column. 

## Todos

- [x] Test in DEV

Tickets:

- BACK-1653

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -99,31 +99,32 @@ model CollectionAuthor {
...
+  name        String            @unique @db.VarChar(255)
+  createdAt   DateTime          @default(now()) @db.DateTime(0)
+  createdBy   String            @db.VarChar(255)
+  updatedAt   DateTime          @updatedAt
+  updatedBy   String?           @db.VarChar(255)
+  collections CollectionLabel[]
+
 
   @@unique([externalId])
   @@index([id])
```
Comment (nina-py) [inline prisma/schema.prisma:113]: Since this is a refactor PR already, it would be good to remove this - we don't need to set up an index on a primary key field, it comes for free.

File: prisma/schema.prisma [hunk 2]
```diff
@@ -99,31 +99,32 @@ model CollectionAuthor {
...
   @@index([name])
-
 }
 
 model CollectionLabel {
-  id                    Int @id @default(autoincrement())
```
Comment (nina-py) [inline prisma/schema.prisma:117]: I think we should keep this field and use it instead of the composite unique ID proposed in this PR. We tried this approach earlier, I think with `CollectionPartnership` entity and Prisma did not take it well at the time (I wish I could remember the details! Types I think were the main problem) - things might have changed so you could still go ahead and try it if you like, but this would be the first place to look if things don't work as expected.

File: prisma/seed.ts [hunk 3]
```diff
@@ -51,12 +54,17 @@ async function main() {
     author: katerina,
     curationCategory: curationCategory1,
   });
+
+  // CollectionLabel table input data
+  const collectionLabelInputData: CreateCollectionLabelInput = {
+    collectionId: katerinaCollection.id,
```
Comment (katerinachinnappan) [inline prisma/seed.ts:60]: i am a bit confused why the `id` field is used to track the `label` and `collection` instead of the `externalId`?

Comment (katerinachinnappan) [reply prisma/seed.ts:60]: this also brings up the question why have both `id` and `externalId` and not keep just one?

Comment (Herraj) [reply prisma/seed.ts:60]: Good question @katerinachinnappan! We use this approach because we don't want the database id i.e `id` to be exposed to the clients/graphql. So we provide an `externalId` field for them to able to fetch specific item via `externalId`.

The internal `id` is then only used for joins/foreign keys. Since this table is merely a mapping table and used to join the other two tables, we remove the `externalId`.

Hope that makes sense!

File: prisma/schema.prisma (hunk 2 above)
Comment (jpetto) [reply prisma/schema.prisma:117]: @nina-py do you remember in what scenarios this might have caused problems before? my mind is unfortunately blank on this 😄.

my vote is to try with the composite id. we should get feedback pretty early when we test the associated resolvers. it would be nice to not have an extra `id` field - particularly based on the intended usage of this table, which will likely have a lot of deletes and inserts, but zero updates.

File: prisma/seed.ts (hunk 3 above)
Comment (jpetto) [reply prisma/seed.ts:60]: good questions!

it's a bad security practice to let auto-incrementing ids out into the world. potentially an attacker could enumerate your database this way. the `externalId` property is our mitigation strategy here (which is why it's used everywhere _except_ internal entity relationships).

at the design of this system, we decided to use auto-incrementing keys for internal relationships. it's more human-readable should someone need to manually inspect relationships, and the cross-table storage is slightly reduced. we didn't hold this position super strongly when designing, but decided to go with it. these kinds of decisions are always up for debate though.

tl;dr - `externalId` could be our one and only identifying field if we felt strongly.

File: prisma/schema.prisma (hunk 2 above)
Comment (jpetto) [reply prisma/schema.prisma:117]: my vote doesn't have to be the winner here though! would love others to chime in as well.

**Review by katerinachinnappan — APPROVED** [review]
